rclcpp/include/rclcpp/node.hpp

```diff
@@ -321,6 +321,24 @@ class Node : public std::enable_shared_from_this<Node>
     ParameterT & value,
     const ParameterT & alternative_value) const;
 
+  /// Get the parameter value; if not set, set the "alternative value" and store it in the node.
+  /**
+   * If the parameter is set, then the "value" argument is assigned the value
+   * in the parameter.  If the parameter is not set, then the "value" argument
+   * is assigned the "alternative_value", and the parameter is set to the
+   * "alternative_value" on the node.
+   *
+   * \param[in] name The name of the parameter to get.
+   * \param[out] value The output where the value of the parameter should be assigned.
+   * \param[in] alternative_value Value to be stored in output and parameter if the parameter was not set.
+   */
+  template<typename ParameterT>
+  void
+  get_parameter_or_set_default(
```

Comment:
What's the difference between this and the function above?

Reply:
I probably should have been more clear in the initial description.  `get_parameter_or` gets the parameter if it exists, and stores it in the output value.  If it doesn't exist, it stores the alternative value in the output value.  In all cases, it doesn't change the parameter existence on the node (so `ros2 param list` would not show the parameter in the list).  `get_parameter_or_set_default` gets the parameter if it exists, and stores it in the output value.  If it doesn't exist, it stores the alternative value in the output value.  In all cases, the parameter is guaranteed to exist on the node after the call, either with the previously set value, or with the new default value.  Does that make sense?

Reply:
I didn't read the docstring close enough to get the distinction, but this certainly makes sense.